sdks/java/io/kafka/src/main/java/org/apache/beam/sdk/io/kafka/KafkaUnboundedReader.java

```diff
@@ -614,7 +619,7 @@ private void nextBatch() {
     partitionStates.forEach(p -> p.recordIter = records.records(p.topicPartition).iterator());
 
     // cycle through the partitions in order to interleave records from each.
-    curBatch = Iterators.cycle(new LinkedList<>(partitionStates));
+    curBatch = Iterators.cycle(new ArrayList<>(partitionStates));
```

Comment:
Was this also recommended by ErrorProne?

Reply:
Yes. 

> `LinkedList` almost never out-performs `ArrayList` or `ArrayDeque`. If you are using `LinkedList` as a list, prefer `ArrayList`. If you are using `LinkedList` as a stack or queue/deque, prefer `ArrayDeque`.

Because `ArrayDeque` rejects nulls I erred on the side of caution

Reply:
We are using LinkedList because of elements deleted in random order (in `advance()`). There are no nulls.  Was it an ErrorProne error?
Overall I don't think the difference matters at all either way. I would be surprised if ErrorProne insisted we change it.

Reply:
Error prone reports it as warning but we now fail builds on warning (unless suppressed of course). Thanks for confirming no nulls.